### PR TITLE
ENH: Add interp() classmethod for ABCPolyBase and implement for Chebyshev

### PR DESCRIPTION
**Pull request by ghost**
see [#9186](https://github.com/numpy/numpy/pull/9186)

A classmethod has been added to the ABCPolyBase base class for polynomial interpolation. A working implementation has already been added for the Chebyshev class, and it is set up to be easily extensible for other polynomial classes.

Also standardized some spacing in the docstrings of functions in _polybase.py and chebyshev.py. Not much, that was all I noticed while working on the main stuff.

### CODE REVIEW

File: numpy/core/include/numpy/numpyconfig.h [hunk 1]
```diff
@@ -35,6 +35,9 @@
 #define NPY_1_11_API_VERSION 0x00000008
 #define NPY_1_12_API_VERSION 0x00000008
 #define NPY_1_13_API_VERSION 0x00000008
+<<<<<<< HEAD
```
Comment (bashtage) [inline numpy/core/include/numpy/numpyconfig.h]: Merge conflicts

File: pavement.py [hunk 2]
```diff
@@ -102,8 +102,13 @@
 RELEASE_NOTES = 'doc/release/1.14.0-notes.rst'
 
 # Start/end of the log (from git)
+<<<<<<< HEAD
```
Comment (bashtage) [inline pavement.py]: Merge conflicts

**Comment by bashtage** [comment]
Should rebase to remove commit noise.

**Comment by uoaei** [comment]
Whoops...will do this tomorrow.

**Comment by charris** [comment]
Might want to read `doc/source/dev/gitwash/development_workflow.rst` if you haven't already.

**Comment by eric-wieser** [comment]
Closing in favor of #9198

**Pull request closed without merging**